rest-server/admin/src/main/scala/com/actionml/admin/MongoAdministrator.scala

```diff
@@ -106,6 +106,22 @@ class MongoAdministrator extends Administrator with JsonParser with Mongo {
     }
   }
 
+  override def updateEngine(json: String): Validated[ValidateError, String] = {
+    parseAndValidate[GenericEngineParams](json).andThen { params =>
+      val existingEngineOpt = engines.get(params.engineId)
+      if (existingEngineOpt.isDefined) { // found the engine to update
+        // re-initialize
+        logger.trace(s"Re-initializing engine for resource-id: ${params.engineId} with new params $json")
+        val query = MongoDBObject("engineId" -> params.engineId)
+        val update = MongoDBObject("$set" -> MongoDBObject("engineFactory" -> params.engineFactory, "params" -> json))
+        enginesCollection.findAndModify(query, update)
+        existingEngineOpt.get.init(json, deepInit = false).andThen(_ => Valid(json))
```

Comment:
This is the same as adding or instantiating and engine from persisted data **except** the initialization is not "deep". This is interpreted by the Engine, but usually means not do re-init the algorithm since it is running and you cannot alter params for an existing model, the engine needs to be deleted along with it's model and re-added to init from a clean starting point.